src/main/java/fr/cnes/sonar/plugins/icode/check/ICodeSensor.java

```diff
@@ -112,17 +110,16 @@ public void execute(SensorContext sensorContext) {
             final String outputOption = "-o";
             final String command = String.join(" ", executable, String.join(" ",files), outputOption, outputPath);
 
-            LOGGER.info("running i-Code CNES and generating results to "+ outputPath);
-            LOGGER.debug("command : " + command);
+            LOGGER.info("Running i-Code CNES and generating results to "+ outputPath);
             try {
                 final Process icode =  Runtime.getRuntime().exec(command);
                 int success = icode.waitFor();
-                    if(success==0){
-                        LOGGER.info("Auto-launch successfully executed i-Code CNES");
-                    }else{
-                        LOGGER.error("i-Code CNES auto-launch analysis failed with exit code "+success);
-                    }
-            } catch (InterruptedException | IOException e) {
+                if(0!=success){
+                    final String message = String.format("i-Code CNES auto-launch analysis failed with exit code %d.",success);
+                    throw new ICodeException(message);
+                }
+                LOGGER.info("Auto-launch successfully executed i-Code CNES.");
+            } catch (InterruptedException | IOException | ICodeException e) {
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: Either re-interrupt this method or rethrow the "InterruptedException". ([squid:S2142](https://sonarcloud.io/organizations/lequal/rules#rule_key=squid%3AS2142))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=fr.cnes.sonarqube.plugins%3Asonaricode&pullRequest=32&issues=AWXW4RBiashydt7kNclZ&open=AWXW4RBiashydt7kNclZ)

[ISSUE_KEY]: AWXW4RBiashydt7kNclZ